### PR TITLE
rewrite README to provide high-level overview and more detailed instructions

### PR DESCRIPTION
**Pull request by mojavelinux**

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -1,45 +1,396 @@
...
+:uri-nvm: https://github.com/creationix/nvm
+:uri-nvm-install: https://github.com/creationix/nvm#installation
+:uri-hbs: http://handlebarsjs.com
+:uri-git: https://git-scm.com
+
+This repository builds and packages the default theme for the MuleSoft documentation site.
+It contains the theme artifacts (page templates, CSS, JavaScript, etc.) and a build.
+The build is used either to preview the theme locally (featuring live updates) or package and publish it to git for consumption by the site generator.
+
+This guide covers the files that make up a theme, identifies prerequisites you need to get started, shows you how to run the preview server to develop on the theme, and, finally, explains the process by which new versions get packaged and published for "`production`" usage.
```
Comment (graphitefriction) [inline README.adoc]: This guide:

* describes the files that make up a theme
* identifies prerequisites you need to get started
* shows you how to run the preview server
* outlines how to version, package and publish a theme

File: README.adoc [hunk 2]
```diff
@@ -1,45 +1,396 @@
...
+  header-menu.css
+  main.css
+  navigation.css
+  primary-content.css
+  scrollbars.css
+  theme.css
+....
+
+A Gulp build is used to compile and assemble these files to produce a distributable theme.
+When the files are built, they are assembled inside the the [.path]_build_ directory, which is safe to remove.
```
Comment (graphitefriction) [inline README.adoc]: delete extra "the"

File: README.adoc [hunk 3]
```diff
@@ -1,45 +1,396 @@
...
+
+ $ git --version
+
+If not, https://git-scm.com/downloads[download and install] the git package for your system.
+
+Next, make sure that you have Node 7.8.0 or better installed.
+While you can install Node from the official packages, we strongly recommend that you use {uri-nvm}[nvm] (Node Version Manager) to install and manage Node.
+Follow the {uri-nvm-install}[installation instructions] to set up nvm on your machine.
+
+One you have installed nvm, open a new terminal and install the stable version of Node using the following command:
```
Comment (graphitefriction) [inline README.adoc]: Once - not one

File: README.adoc [hunk 4]
```diff
@@ -1,45 +1,396 @@
...
 
-[source, bash]
-yarn install
+The stylesheets are written in CSS.
+These stylesheets utilize CSS variables to keep the CSS DRY and easy to customize.
 
-== Building the theme
+Within this project, the files are separated into modules to help organize the rules and make them easier to find.
+These files get combined (and minified) into a single file by the theme build, named [.path]_theme.css_.
+At the same time, the CSS is enhanced using postcss in much the same was as a CSS preprocessor works, only the modifications are made to the CSS directly.
```
Comment (graphitefriction) [inline README.adoc]: the same way - not was

File: README.adoc [hunk 5]
```diff
@@ -1,45 +1,396 @@
...
+This is the main configuration file for the build.
+It defines the path where the files are assembled when built, which defaults to the [.path]_build_ folder.
+It also defines the path where the theme assets will reside in the production site, which defaults to [.path]__theme_.
 
-== Building the preview page
+== Publishing a Release
 
-In order to preview the theme without doing a full build of the site, this project includes a fake static page: `preview-site/index.html`.
-You can build this static page with the templates included in the theme with this:
+Once you're done making changes to the theme and want to role out those changes, you'll need to publish a release.
```
Comment (graphitefriction) [inline README.adoc]: roll out - not role

File: README.adoc [hunk 6]
```diff
@@ -1,45 +1,396 @@
...
+That task creates a git tag{blank}footnote:[Tag names are sequential, so each tag uses a number that is one greater than the previous one (e.g., from v9 to v10).], runs the theme build, bundles the theme as a zip file, and uploads it to the release page (which is associated with that newly created tag).
 
-== Development setup
+=== Configuring the Release Task
 
-If you work on the theme assets (CSS, JS, fonts...), it's often better to avoid building the preview manually.
-This project includes a small HTTP server and a live build-reload setup.
-You just have to run this:
+The CI job (environment and script) is configured in [.path]_.travis.yml_.
+After running through the set up process described above, it runs the `gulp release` command.
```
Comment (graphitefriction) [inline README.adoc]: setup - not set up

File: README.adoc [hunk 7]
```diff
@@ -1,45 +1,396 @@
...
 
-Once this is running in your terminal, each change to stylesheets, scripts, templates... will trigger a reload in your browser.
-You can browser the preview page here: `http://localhost:8080`.
+repository.owner:: The GitHub organization where the main repository (not a fork) is hosted.
+repository.name:: The name of the repository on GitHub.
+GITHUB_TOKEN:: The authentication token of the release user, granting write access to the CI job.
 
-== Creating a release
+The first two properties, `repository.owner` and `repository.name`, are defined in the [.path]_theme.yml_ file.
+The last property, `GITHUB_TOKEN`, is defined on the {uri-ci}/settings[settings page] for the CI job.
```
Comment (graphitefriction) [inline README.adoc]: This section needs to specify the default behavior, which is only implied whereas the manual override takes center stage.

File: README.adoc [hunk 8]
```diff
@@ -1,45 +1,396 @@
...
+:uri-gulp: http://gulpjs.com
+:uri-yarn: https://yarnpkg.com
+:uri-nvm: https://github.com/creationix/nvm
+:uri-nvm-install: https://github.com/creationix/nvm#installation
+:uri-hbs: http://handlebarsjs.com
+:uri-git: https://git-scm.com
+
+This repository builds and packages the default theme for the MuleSoft documentation site.
+It contains the theme artifacts (page templates, CSS, JavaScript, etc.) and a build.
+The build is used either to preview the theme locally (featuring live updates) or package and publish it to git for consumption by the site generator.
```
Comment (mojavelinux) [inline README.adoc]: Done.

File: README.adoc (hunk 1 above)
Comment (mojavelinux) [inline README.adoc]: Done.

File: README.adoc (hunk 2 above)
Comment (mojavelinux) [inline README.adoc]: Done.

File: README.adoc (hunk 3 above)
Comment (mojavelinux) [inline README.adoc]: Done.

File: README.adoc (hunk 4 above)
Comment (mojavelinux) [inline README.adoc]: Done.

File: README.adoc (hunk 5 above)
Comment (mojavelinux) [inline README.adoc]: Done.

File: README.adoc (hunk 6 above)
Comment (mojavelinux) [inline README.adoc]: Done.

File: README.adoc (hunk 7 above)
Comment (mojavelinux) [inline README.adoc]: Done.

**Comment by mojavelinux** [comment]
Changes incorporated!

**Pull request merged**